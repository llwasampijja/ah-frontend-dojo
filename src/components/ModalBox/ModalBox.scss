* {
  margin: 0;
  padding: 0;
}

// ModalBox backdrop styles
.backdrop {
  position: fixed;
  height: 100vh;
  background: rgba(0, 0, 0, 0.5);
  top: 0;
  left: 0;
  width: 100%;
  z-index: -1;
}

// ModalBox styles
.modal {
  width: 40rem;
  box-shadow: 0 0.2rem 0.8rem rgba(0, 0, 0, 0.26);
  position: fixed;
  top: 20vh;
  left: 5rem;
  
  // Modal header styles
  &__header {
    padding: 1rem;
    background: #112387;
    text-align: center;

    // modal header title styles
    &__title {
      color: #ffffff;
      display: inline-block;
    }

    // modal header close button styles
    &__btn {
      color: #ff0000;
      float: right;
      padding: 0.3rem;
    }
  }
}
