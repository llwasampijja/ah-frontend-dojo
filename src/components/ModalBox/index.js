// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

// styles
import 'components/ModalBox/ModalBox.scss';

const ModalBox = (WrappedComponent, props) => (innerFunctionProps) => {
  // props destructuring
  const { title, closeModal } = innerFunctionProps;

  return (
    <div>
      {/* backdrop */}
      <div
        className="backdrop"
      />

      {/* ModalBox */}
      <div id="modal" className="modal">
        <header className="modal__header">
          <h4 className="modal__header__title">{title}</h4>
          <button
            className="modal__header__btn"
            onClick={closeModal}
            type="button"
          >
            X
          </button>
        </header>
        <section className="modal__section">
          <WrappedComponent {...props} />
        </section>
      </div>
    </div>
  );
};

// props validation
ModalBox.propTypes = {
  title: PropTypes.string,
  closeModal: PropTypes.func,
};

// props default values
ModalBox.defaultProps = {
  title: 'Form Modal',
};

export default ModalBox;
