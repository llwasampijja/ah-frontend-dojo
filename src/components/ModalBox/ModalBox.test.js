// react libraries
import React from 'react';

// third party libraries
import { shallow, mount } from 'enzyme';

// components
import ModalBox from 'components/ModalBox';

// test ModalBox component
describe('ModalBox Component', () => {
  it('matches the snapshot', () => {
    const component = shallow(<ModalBox />);
    expect(component).toMatchSnapshot();
  });
});

// test use-case for ModalBox component
describe('Wrapped Components', () => {
  const props = {
    title: 'test',
    onClose: jest.fn(),
  };

  it('renders component', () => {
    const testWrappedComponent = () => <div><h1>Test</h1></div>;
    const TestRenderedComponent = ModalBox(testWrappedComponent);
    const calledComponent = mount(<TestRenderedComponent onClose={props.onClose} />);
    const wrapper = shallow(<TestRenderedComponent {...props} />);
    const mockCloseFunction = jest.fn();

    // test use-case for the title of ModalBox
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.modal__header__title').text()).toBe('test');

    // test use-case for onclick of button
    calledComponent.find('button').simulate('click');
    calledComponent.onClose = mockCloseFunction;
    calledComponent.onClose();
    expect(mockCloseFunction).toHaveBeenCalled();
    expect(calledComponent.onClose).toHaveBeenCalled();
  });
});
