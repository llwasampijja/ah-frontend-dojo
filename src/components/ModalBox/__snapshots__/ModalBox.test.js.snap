// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBox Component matches the snapshot 1`] = `<Component />`;

exports[`Wrapped Components renders component 1`] = `
<div>
  <div
    className="backdrop"
  />
  <div
    className="modal"
    id="modal"
  >
    <header
      className="modal__header"
    >
      <h4
        className="modal__header__title"
      >
        test
      </h4>
      <button
        className="modal__header__btn"
        type="button"
      >
        X
      </button>
    </header>
    <section
      className="modal__section"
    >
      <testWrappedComponent />
    </section>
  </div>
</div>
`;
